packages/react-router/lib/rsc/browser.tsx

```diff
@@ -288,17 +290,23 @@ export function getRSCSingleFetchDataStrategy(
       context.set(renderedRoutesContext, []);
       let results = await dataStrategy(args);
       // patch into router from all payloads in map
-      const renderedRouteById = new Map(
-        context.get(renderedRoutesContext).map((r) => [r.id, r])
-      );
+      const renderedRoutesById = new Map<string, RenderedRoute[]>();
```

Comment:
TODO: Add a comment about investigating why we have multiple rendered routes by ID.